### PR TITLE
fix(format): Use yaml-language-server formatter so that comments aren't removed

### PR DESCRIPTION
**Pull request by abdelnn**
## Issue #, if available:
Related issue in aws-toolkit-vscode: https://github.com/aws/aws-toolkit-vscode/issues/2603
Taking over from PR #82 

## Description of changes:
This PR provides the minimal changes to fully support the yaml-language-server formatter instead of the js-yaml one. This provides us a few features:

- Comments are no longer removed when the document is formatted
- FormatterOptions from the LSP are fully respected when formatting

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by wong-a — DISMISSED** [review]

File: src/tests/yamlFormat.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# this is my comment
+hello: world # this is another comment
+`
+        const formattedTextEdits = await getFormat(yaml, Range.create(Position.create(0, 0), Position.create(yaml.length, yaml.length)), {
+            tabSize: 4,
+            insertSpaces: true
+        })
+
+        assert.equal(formattedTextEdits.length, 0, 'Expected 0 text edits when formatting input YAML')
+    })
```
Comment (jormello) [inline src/tests/yamlFormat.test.ts:30]: Can we add at least one test case with a non-zero amount of expected text edits? If I recall correctly, the previous PR had an issue where it no longer produced any text edits. This wasn't initially caught since unit test only covered the no-op scenario.

**Review by jormello — CHANGES_REQUESTED** [review]

File: src/tests/yamlFormat.test.ts (hunk 1 above)
Comment (abdelnn) [reply src/tests/yamlFormat.test.ts:30]: Will add one

**Review by jormello — APPROVED** [review]

**Review by wong-a — APPROVED** [review]
